### PR TITLE
[Aptos Compression] Small cleanups and metrics improvements

### PR DESCRIPTION
**Pull request by JoshLind**
### Description
This PR offers several small cleanups and metrics improvements to the `aptos-compression` crate (each in their own commit):
1. Move the compression client to a separate file. This prevents us from having to expose the metrics file globally.
2. Add a new error type so that we can differentiate between compression and decompression errors. 
3. Several small code cleanups and metric improvements. This includes: (i) adding metrics for sent and received messages (not just sent messages, as is being tracked today); (ii) better histogram buckets for tracking serialization and compression; and (iii) better metrics for tracking compression and decompression errors.

### Test Plan
New and existing test infrastructure.

### CODE REVIEW

File: crates/aptos-compression/src/metrics.rs [hunk 1]
```diff
@@ -62,35 +39,89 @@ pub static OPERATION_LATENCY: Lazy<HistogramVec> = Lazy::new(|| {
...
+
+/// Updates the operation metrics based on the given data
+/// (e.g., raw and compressed data sizes).
+fn update_operation_metrics(
+    operation: &str,
+    client: &CompressionClient,
+    raw_data: &Vec<u8>,
+    compressed_data: &Vec<u8>,
+) {
+    increment_compression_byte_count(operation, RAW_BYTES, client, raw_data.len() as u64);
```
Comment (bchocho) [inline crates/aptos-compression/src/metrics.rs:120]: Wonder if the compression ratio is an interesting metric as well? I see you've removed it (although it was only used for trace).

(No strong opinion here, it just crossed my mind)

**Review by bchocho — APPROVED** [review]
LGTM

File: crates/aptos-compression/src/metrics.rs (hunk 1 above)
Comment (JoshLind) [reply crates/aptos-compression/src/metrics.rs:120]: Aah, we're still tracking the ratio -- the logic I removed was for the `trace!` logs, which we never use. In grafana we're tracking the ratio by dividing the raw bytes by the compressed bytes, so nothing should be lost 😄

Comment (bchocho) [reply crates/aptos-compression/src/metrics.rs:120]: Oh ok, I was actually thinking of possible outliers -- if some data compresses too well or not well at all.

**Review by brianolson — APPROVED** [review]

**Pull request merged**